src/core.jl

```diff
@@ -118,13 +137,66 @@ function savefile(file::TextFile, content)
     write(file.filename, content)
 end
 
+#######################################
+# Preprocess
+
+preprocess(::File, actual; kw...) = actual
+
+# plain TXT
+preprocess(::File{format"TXT"}, actual; kw...) = string(actual)
+preprocess(::File{format"TXT"}, actual::Number; kw...) = actual
+function preprocess(::File{format"TXT"}, actual::AbstractArray{<:AbstractString}; kw...)
+    return join(actual, '\n')
+end
+
+# SHA256
+preprocess(::File{format"SHA256"}, actual; kw...) = bytes2hex(sha256(string(actual)))
+
+# images
+function preprocess(
+    ::File{format"TXT"}, actual::AbstractArray{<:Colorant};
+    size = (20,40), kw...)
+
+    strs = @withcolor ImageInTerminal.encodeimg(
+        ImageInTerminal.SmallBlocks(),
+        ImageInTerminal.TermColor256(),
+        actual,
+        size...)[1]
+    str = join(strs,'\n')
+end
+
+function preprocess(::File{format"SHA256"}, actual::AbstractArray{<:Colorant}; kw...)
+    size_str = bytes2hex(sha256(reinterpret(UInt8,[map(Int64,size(actual))...])))
+    img_str = bytes2hex(sha256(reinterpret(UInt8,vec(rawview(channelview(actual))))))
+
+    return size_str * img_str
+end
+
+
 ##########################################
 
 # Final function
 # all other functions should hit one of this eventually
 # Which handles the actual testing and user prompting
 
-function _test_reference(equiv, rendermode, file::File, actual::T) where T
+function test_reference(
+    filename::AbstractString, actual;
+    by = nothing, render = nothing, kw...)
+
+    test_reference(query_extended(filename), actual, render, by; kw...)
+end
+
+function test_reference(
+    file::File,
+    actual::T,
+    rendermode::Union{RenderMode, Nothing}=nothing,
+    equiv::Union{Function, Nothing}=nothing;
+    kw...) where T
```

Comment:
Arguments order is changed because I made `rendermode` and `equiv` optional.

`kw...` are all passed to `preprocess` as it was (the only possible `kw` here is `size` for image type). It might be changed in the future, so I didn't document it.